doc/doxygen/public/developer-faq.doxygen

```diff
@@ -0,0 +1,291 @@
+ <li>Call:</li>
+ @code{sh}
+ php tools/create_test.php /BUILD_DIRECTORY/ /PATH_TO_HEADER/MyClass.h \
+  "FIRSTNAME LASTNAME" > ./src/tests/class_tests/openms/source/MyClass_test.cpp
+ @endcode
+ </ol>
+
+ @subsection how_to_add_a_new_gui_test How to add a new GUI test for a class
+
+ Suppose you want to create a GUI class for the class called <code>MyClass</code>.
+ To add a new GUI test:
+ <ol>
+ <li>Create the <code>MyClass_test.cpp</code> in <code>src/tests/class_tests/openms_gui/source</code>.</li>
+ <li>Add it to <code>src/tests/class_tests/openms_gui/CMakeLists.txt</code> in the GUI section.</li>
+ <li>Have a look at existing GUI tests, as they use the QT TestLib framework and not the OpenMS macros.</li>
+ </ol>
+
+ <h2><a class="anchor" id="when_executing_make_test_all_tests_fail"></a>All tests fail when executing <code>make_test</code></h2>
+
+ Check the <code>LD_LIBRARY_PATH</code> environment variable:
+
+ Print the <code>LD_LIBRARY_PATH</code> with <code>echo $LD_LIBRARY_PATH</code>. If your <code>/lib/</code> folder is included, check that <code>libOpenMS.so</code> is present. With the <code>ldd</code> command, you can show the libraries used by an executable, e.g. <code>ldd /bin/ClassTest_test</code>.
+
+ @section debugging Debugging
+
+ The following section provides information on how to debug your code.
+
+ @subsection how_to_find_out_which How to find out which shared libraries are used by an application
+
+ Linux: Use <code>ldd</code>.
+
+ Windows (Visual studio console): See <a href="http://www.dependencywalker.com/">Dependency Walker</a> (use x86 for 32 bit builds and the x64 version for 64bit builds. Using the wrong version of depends.exe will give the wrong results) or
+ <code>dumpbin /DEPENDENTS OpenMS.dll</code>.
+
+ @subsection how_to_get_a_list How to get a list of the symbols defined in a (shared) library or object file?
+
+ Linux: Use <code>nm <library></code>.
+
+ Use <code>nm -C</code> to switch on demangling of low-level symbols into their C++-equivalent names. <code>nm</code> also accepts <code>.a</code> and <code>.o</code> files.
+
+ Windows (Visual studio console): Use <code>dumpbin /ALL <library></code>.
+
+ Use dumpbin on object files (.o) or (shared) library files (.lib) or the DLL itself e.g. <code>dumpbin /EXPORTS OpenMS.dll</code>.
+
+ @subsection how_to_profile_code How to profile code
+
+ IBM's profiler, available for all platforms (and free for academic use): Purify(Plus) and/or Quantify.
+
+ Windows: this is directly supported by Visual Studio (Depending on the edition: Team and above). Follow their documentation.
+
+ Linux:
+ <ol>
+ <li>Build OpenMS in debug mode (set <code>CMAKE_BUILD_TYPE</code> to <code>Debug</code>).</li>
+ <li>Call the executable with valgrind: <code>valgrind –tool=callgrind</code>.</li>
+ @warning Other processes running on the same machine can influence the profiling. Make sure your application gets enough
```

Comment:
use `@warning` instead of `\warning` for consistency.

Reply:
I've been using @warning the whole time.